File: java/src/main/java/io/fivexer/sdk/model/Team.java
package io.fivexer.sdk.model;

/**
 * A team: a routing primitive, not a label — matching sees the derived {@code tag}.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class Team {
    private String id;
    private String key;
    private String tag;
    private String name;
    private String description;
    private String color;
    private String createdAt;

    public String getId() { return id; }
    public String getKey() { return key; }
    public String getTag() { return tag; }
    public String getName() { return name; }
    public String getDescription() { return description; }
    public String getColor() { return color; }
    public String getCreatedAt() { return createdAt; }
}

File: php/src/Model/Team.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A team: a routing primitive, not a label — matching sees the derived `tag`.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class Team
{
    public function __construct(
        public readonly string $id,
        public readonly string $key,
        /** The routing tag the server derived from the key. */
        public readonly string $tag,
        public readonly string $name,
        public readonly ?string $description,
        public readonly ?string $color,
        /** ISO-8601 */
        public readonly string $createdAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            key: (string) ($data['key'] ?? ''),
            tag: (string) ($data['tag'] ?? ''),
            name: (string) ($data['name'] ?? ''),
            description: isset($data['description']) ? (string) $data['description'] : null,
            color: isset($data['color']) ? (string) $data['color'] : null,
            createdAt: (string) ($data['createdAt'] ?? ''),
        );
    }
}
